contrib/dockerd-rootless.sh

```diff
@@ -118,5 +118,13 @@ else
 		# https://github.com/moby/moby/issues/41230
 		chcon system_u:object_r:iptables_var_run_t:s0 /run
 	fi
+
+	# Workaround for "x509: certificate signed by unknown authority" on openSUSE Tumbleweed.
+	# https://github.com/rootless-containers/rootlesskit/issues/225
+	realpath_etc_ssl=$(realpath /etc/ssl)
```

Comment:
I'm slightly confused here; from that ticket, it looks like the files _inside_ `/etc/ssl` are symlinks, but `/etc/ssl` itself is not, correct?

```bash
$ docker run --rm opensuse/tumbleweed sh -c 'ls -l /etc | grep ssl && ls -l /etc/ssl'
drwxr-xr-x  2 root root     4096 Jun  4 13:11 ssl
total 0
lrwxrwxrwx 1 root root 43 May  8 20:10 ca-bundle.pem -> ../../var/lib/ca-certificates/ca-bundle.pem
lrwxrwxrwx 1 root root 33 May  8 20:10 certs -> ../../var/lib/ca-certificates/pem

$ docker run --rm opensuse/tumbleweed realpath /etc/ssl
/etc/ssl
```


Isn't the issue that the files _inside_ it cannot be used (because they're symlinked?)

Reply:
In other words, would we:

- find `/var/lib/ca-certificates/` (based on one of the symlinks inside the directory)
- then `mount --rbind /var/lib/ca-certificates/ /etc/ssl` ?

Currently it looks like it's just mounting `/etc/ssl` on `/etc/ssl`

(perhaps I'm wrong!)

Reply:
> it looks like the files inside /etc/ssl are symlinks, but /etc/ssl itself is not, correct?

true

> Isn't the issue that the files inside it cannot be used (because they're symlinked?)

Yes.
The issue is that `ca-bundle.pem -> ../../var/lib/ca-certificates/ca-bundle.pem` is no longer accessible as it is now resolved into `/etc/var/lib/ca-certificates/ca-bundle.pem, not `/var/lib/ca-certificates/ca-bundle.pem`, because the base directory is now `/etc/.ro780432485/ssl`, not `/etc/ssl`

> find /var/lib/ca-certificates/

No, that probably doesn't work for other distros.

> Currently it looks like it's just mounting /etc/ssl on /etc/ssl

It is mounting `/etc/ssl` in the parent namespace on `/etc/ssl` in the child namespace.

Reply:
But what's the `realpath_etc_ssl=$(realpath /etc/ssl)` doing? As there's no symlink to resolve in that case.

So the effective change is that we delete `/etc/ssl` (and the symlinks) and replace it with an empty directory?

Reply:
`rootlesskit --copy-up=/etc` mounts an empty tmpfs on `/etc` (to make it writable), and simlinkify `/etc/ssl` to `/etc/.ro780432485/ssl`.

So `realpath_etc_ssl` = `/etc/.ro780432485/ssl`.

Reply:
Ahhhhhh! That's the bit I was missing! Sorry for the confusion 🤗 ❤️